python/ray/serve/_private/proxy_response_generator.py

```diff
@@ -95,6 +95,16 @@ async def __anext__(self):
 
             if self._result_callback is not None:
                 result = self._result_callback(result)
+        except asyncio.CancelledError as e:
+            # This cancellation can happen from client dropped connection before the
+            # request is completed. If self._response is not already cancelled, we want
+            # to explicitly cancel the task, so it doesn't waste cluster resource in
+            # this case and can be terminated gracefully.
```

Comment:
Can you specify that this is only the case for gRPC?

This is also causing the two protocols to again diverge in behavior. Perhaps it's better if we were to unify them by pulling the `disconnect_task` logic out of this layer and instead waiting for it at a higher layer, then calling cancelling the task when it finishes. Not a big deal, but better for them to behave as similarly as possible.

Reply:
ok, ,let me update the comment. Will create a follow up ticket to pull out the disconnect task logic!

Reply:
follow up issue created https://github.com/ray-project/ray/issues/40796